tests/test_transformation.py

```diff
@@ -146,3 +149,63 @@ def test_does_pdf_contain_cmyk(client, data, result):
 )
 def test_does_pdf_contain_rgb(client, data, result):
     assert does_pdf_contain_rgb(BytesIO(data)) == result
+
+
+def detect_color_space(page):
+    if does_pdf_contain_cmyk(page):  # CMYK images have 4 color channels
+        return "CMYK"
+    return "RGB"
+
+
+def compare_images(page1, page2):
+    pix_map1 = page1.get_pixmap()
+    pix_map2 = page2.get_pixmap()
+
+    img1 = Image.frombytes("RGB", [pix_map1.width, pix_map1.height], pix_map1.samples)
+    img2 = Image.frombytes("RGB", [pix_map2.width, pix_map2.height], pix_map2.samples)
+
+    diff = ImageChops.difference(img1, img2)
+    return not diff.getbbox()  # If there's no difference, returns True
+
+
+def test_cmyk_pdf_transformation():
+    if not os.environ.get("TEST_CMYK_PDF_ENABLED", False):
```

Comment:
We should have pytest skip this test if `TEST_CMYK_PDF_ENABLED` is False.

The `@pytest.mark.skipif` decorator should work for this.